src/components/forms/OpaqueIngressSecretForm.js

```diff
@@ -0,0 +1,425 @@
+import React, { Component } from "react";
+import { PropTypes } from "prop-types";
+import { IcseFormGroup } from "../Utils";
+import { IcseNameInput, IcseTextInput, IcseToggle } from "../Inputs";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseSelect } from "../Dropdowns";
+import {
+  Tile,
+  DatePicker,
+  DatePickerInput,
+  Tag,
+  TextArea,
+  NumberInput,
+} from "@carbon/react";
+import "./opaque-ingress-secret.css";
+
+const labelColors = [
+  "red",
+  "magenta",
+  "purple",
+  "blue",
+  "cyan",
+  "teal",
+  "green",
+];
+
+class OpaqueIngressSecretForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.data;
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.handleLabels = this.handleLabels.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  /**
+   * handle text input change
+   * @param {event} event
+   */
+  handleInputChange(event) {
+    this.setState(this.eventTargetToNameAndValue(event));
+  }
+
+  /**
+   * handle toggle
+   * @param {event} event
+   */
+  handleToggle(name) {
+    this.setState(this.toggleStateBoolean(name, this.state));
+  }
+
+  /**
+   * handle labels
+   * @param {event} event
+   */
+  handleLabels(event) {
+    let labelList = event.target.value
+      ? event.target.value
+          .replace(/\s\s+/g, "") // replace extra spaces
+          .replace(/,(?=,)/g, "") // prevent null tags from
+          .replace(/[^\w,-]/g, "")
+          .split(",")
+      : [];
+    this.setState({ labels: labelList });
+  }
+
+  render() {
+    let composedId = `opaque-ingress-secrets-from-${this.props.data.name}`;
+    return (
```

Comment:
Username Password Secret & Arbitrary Secret sub forms lose their background color when in the cluster form. However, they have it in the modal. 

<img width="300" alt="Screenshot 2023-08-31 at 11 22 54 AM" src="https://github.com/IBM/icse-react-assets/assets/11401353/4794a7cf-999b-4fad-93c1-295959c33bf7">
<img width="300" alt="Screenshot 2023-08-31 at 11 24 44 AM" src="https://github.com/IBM/icse-react-assets/assets/11401353/e657f7f7-e697-4458-99b5-a9b6b8b3faa8">